opentelemetry-sdk/src/propagation/baggage.rs

```diff
@@ -48,14 +48,17 @@ fn baggage_fields() -> &'static [String; 1] {
 /// }
 ///
 /// // Add new baggage
-/// let cx_with_additions = cx.with_baggage(vec![KeyValue::new("server_id", 42)]);
+/// let mut baggage = Baggage::new();
+/// let _ = baggage.insert("server_id", "42");
+///
+/// let cx_with_additions = cx.with_baggage(baggage);
```

Comment:
can you think any Rust magic like Into impl for a slice/vec of KeyValue for Baggage, so users can write
cx.with_baggage([KeyValue::new("f","b"),...]) or slice or vec...

Reply:
You are still able to do `cx.with_baggage(Baggage::from_iter([KeyValue::new("f","b"),...]))` which is basically the old behaviour.

Reply:
Can't we just take `Into<Baggage>` as parameter and then implement `Into<Baggage>` for `Vec<KeyValue>`?

Reply:
> I think I found a workable solution.

Does it mean we should now change this example to use:
```rust
let cx_with_additions = cx.with_baggage([("server_id", "42")]);
```

Reply:
@lalitb that's not possible for now. it could either be an iterator with `KeyValue` or `KeyValueMetadata` entries. Do we want to support this here, then we probably need to implement an other trait or so.